sickbeard/versionChecker.py

```diff
@@ -583,11 +583,7 @@ def set_newest_text(self):
         # if we're up to date then don't set this
         sickbeard.NEWEST_VERSION_STRING = None
 
-        if self._num_commits_ahead:
-            logger.log(u"Local branch is ahead of " + self.branch + ". Automatic update not possible.", logger.WARNING)
-            newest_text = "Local branch is ahead of " + self.branch + ". Automatic update not possible."
-
-        elif self._num_commits_behind > 0:
+        if self._num_commits_behind > 0 or (sickbeard.GIT_RESET and sickbeard.BRANCH in sickbeard.GIT_RESET_BRANCHES):
```

Comment:
If there's no branches in the list, it should accept as well

```
or (sickbeard.GIT_RESET and (not sickbeard.GIT_RESET_BRANCHES or sickbeard.BRANCH in sickbeard.GIT_RESET_BRANCHES)) 
```

The same logic some lines below (probably a function would fit: def should_reset)